fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/query/plan/synthetic/SyntheticRecordPlannerTest.java

```diff
@@ -619,6 +638,63 @@ public void joinIndex() throws Exception {
             List<Tuple> results3 = recordStore.scanIndex(index, IndexScanType.BY_VALUE, range, null, ScanProperties.FORWARD_SCAN).map(IndexEntry::getKey).asList().join();
             assertEquals(expected3, results3);
         }
+
+        try (FDBRecordContext context = openContext()) {
+            final FDBRecordStore recordStore = recordStoreBuilder.setContext(context).open();
+
+            final QueryPlanner planner = setupPlanner(recordStore, null);
+
+            RecordQuery query = RecordQuery.newBuilder()
+                    .setRecordType("Simple_Other")
+                    .setFilter(Query.field("simple").matches(Query.field("str_value").equalsValue("even")))
+                    .build();
+
+            RecordQueryPlan plan = planner.plan(query);
```

Comment:
Should there be assertions here on the plan (e.g., that it uses the join index)?

Reply:
done.